src/utils/FormattingUtils.ts

```diff
@@ -17,28 +17,33 @@ limitations under the License.
 
 import { ReactElement, ReactNode } from "react";
 
-import { _t } from "../languageHandler";
+import { _t, getCurrentLanguage } from "../languageHandler";
 import { jsxJoin } from "./ReactUtils";
+const locale = getCurrentLanguage();
+
+// It's quite costly to instanciate `Intl.NumberFormat`, hence why we do not do
+// it in every function call
+const compactFormatter = new Intl.NumberFormat(locale, {
+    notation: "compact",
+});
 
 /**
  * formats numbers to fit into ~3 characters, suitable for badge counts
```

Comment:
Might be worth updating the comment to say it also rounds, as shown by the example